src/hooks/useCommunityChart.ts

```diff
@@ -0,0 +1,72 @@
+// };
+import { useEffect, useState } from 'react';
+
+import { IDateStats } from '@/constants/affiliates';
+import { AffiliatesProgramMetric, AffiliatesProgramPeriod } from '@/constants/charts';
+
+import { log } from '@/lib/telemetry';
+
+import { useEndpointsConfig } from './useEndpointsConfig';
+
+export const useCommunityChart = (selectedChartMetric: AffiliatesProgramMetric) => {
+  const { affiliatesBaseUrl } = useEndpointsConfig();
+  // State to keep track of selected period
+  const [selectedPeriod, setSelectedPeriod] = useState<AffiliatesProgramPeriod>(
+    AffiliatesProgramPeriod.PeriodAllTime
+  );
+
+  // State to store the metric data for the currently selected period
+  const [metricData, setMetricData] = useState<{ date: number; cumulativeAmount: number }[]>([]);
+
+  useEffect(() => {
+    fetchMetricData();
+  }, [selectedChartMetric, selectedPeriod]);
+
+  const getStartDate = (): string => {
```

Comment:
[nit] technically does not return a date, `getStartDateISOString`, or we can return the date so functions that consume this value can have greater flexbility on how to use this.


[discussion] also, feels like this should be a general util function. i feel like we have this pattern of looking back 1,7,30 days etc. in several parts of our codebase. if it's a huge pain i don't think we need to implement that now, but maybe a ticket for this would be good?